### PR TITLE
convert machobj.c to D

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/mach.d [hunk 1]
```diff
@@ -341,51 +364,41 @@ enum
...
+
+    /* LITTLE_ENDIAN for x86
+     * uint r_symbolnum:24,
+     *      r_pcrel    :1,
+     *      r_length   :2,
+     *      r_extern   :1,
+     *      r_type     :4;
+     */
+    uint xxx;
+    void r_symbolnum(uint r) { assert(!(r & ~0x00FF_FFFF)); xxx = (xxx & ~0x00FF_FFFF) | r; }
```
Comment (jacob-carlborg) [inline src/dmd/backend/mach.d]: I think this looks pretty ugly on one line.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by ibuclaw** [comment]
Rebase please.

**Comment by WalterBright** [comment]
rebased

**Comment by thewilsonator** [comment]
```
CFISection::cfiParse() unexpected relocation size at r_address=0x00000013
ld: in section __TEXT,__text reloc 0: length != 2 and X86_64_RELOC_SIGNED* not supported file 'checkwhitespace.o' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: linker exited with status 1
make: *** [checkwhitespace] Error 1
```

**Comment by WalterBright** [comment]
Yeah, I have some bisecting to do with the old machobj.c.

**Comment by WalterBright** [comment]
The problem is in Obj_term(), the largest and most complex function :-)

**Comment by WalterBright** [comment]
Fixed Obj_term(). Evidently, that wasn't the only bug. Sigh.

**Comment by thewilsonator** [comment]
```
Running DScanner
--
  | ../generated/linux/release/64/dscanner-3a859d39c4b59822b1bc0452b3ddcd598ef390a2/dsc --config .dscanner.ini --styleCheck dmd -I.
  | dmd/backend/machobj.d(140:60)[warn]: Empty declaration
  | posix.mak:632: recipe for target 'dscanner' failed
```

**Pull request merged**